server/punishments.js

```diff
@@ -42,9 +42,11 @@ const AUTOLOCK_POINT_THRESHOLD = 8;
 
 /**
  * @typedef {object} PunishmentEntry
- * @property {string[]} keys
+ * @property {string[]} ips
+ * @property {string[]} alts
  * @property {string} punishType
- * @property {[number, string]} rest
```

Comment:
We should still retain `rest` as an `any[]` array. It's intended to let the punishment system be extensible.

Reply:
I'm not sure if I completely understood what you mean but I think https://github.com/Zarel/Pokemon-Showdown/pull/5599/commits/e47f4598312b804e988759f4e259c9b07b100bde solves it.